TST: spatial rotation failure in (1.6.3) wheels repo (ARM64)

Log location: https://travis-ci.com/github/MacPython/scipy-wheels/jobs/499313788
Summary below; only observed with ARM64 Python 3.9 (no other Python versions with ARM, so far)

```
=================================== FAILURES ===================================
_________________________ test_align_vectors_no_noise __________________________
[gw3] linux -- Python 3.9.2 /venv/bin/python
/venv/lib/python3.9/site-packages/scipy/spatial/transform/tests/test_rotation.py:992: in test_align_vectors_no_noise
    assert_allclose(rmsd, 0, atol=1e-7)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=1e-07
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference: 1.1920929e-07
E   Max relative difference: inf
E    x: array(1.192093e-07)
E    y: array(0)
        a          = array([[-1.04034169,  2.03198748,  0.13181328],
       [-1.06797688,  1.27831197, -0.1702608 ],
       [ 2.36982519, -0.47848099,  1.87607227],
       [ 0.63945834, -0.84014387,  0.10038155],
       [ 0.40838635,  0.71920354,  0.56158733]])
        b          = array([[ 0.50828076,  0.70253058, -2.11583556],
       [-0.03922247,  0.33711413, -1.63965256],
       [ 1.83370954,  0.94258967,  2.26134842],
       [-0.02364188, -0.21230115,  1.03884283],
       [ 0.86117662,  0.48259327, -0.15777212]])
        c          = <scipy.spatial.transform.rotation.Rotation object at 0xffffa1c236c0>
        est        = <scipy.spatial.transform.rotation.Rotation object at 0xffffa1c23b70>
        rmsd       = 1.1920928955078125e-07
        rnd        = RandomState(MT19937) at 0xFFFFA2A28E40
```

--- a/scipy/spatial/transform/tests/test_rotation.py
+++ b/scipy/spatial/transform/tests/test_rotation.py
@@ -2,7 +2,7 @@ import pytest
 
 import numpy as np
 from numpy.testing import assert_equal, assert_array_almost_equal
-from numpy.testing import assert_allclose, assert_array_less
+from numpy.testing import assert_allclose
 from scipy.spatial.transform import Rotation, Slerp
 from scipy.stats import special_ortho_group
 from itertools import permutations
@@ -725,7 +725,7 @@ def test_inv_single_rotation():
     assert_array_almost_equal(res1, eye)
     assert_array_almost_equal(res2, eye)
 
-    x = Rotation.from_quat(np.random.normal(size=(1, 4)))
+    x = Rotation.from_quat(rnd.normal(size=(1, 4)))
     y = x.inv()
 
     x_matrix = x.as_matrix()
@@ -753,7 +753,7 @@ def test_single_identity_magnitude():
 
 def test_identity_invariance():
     n = 10
-    p = Rotation.random(n)
+    p = Rotation.random(n, random_state=0)
 
     result = p * Rotation.identity(n)
     assert_array_almost_equal(p.as_quat(), result.as_quat())
@@ -764,7 +764,7 @@ def test_identity_invariance():
 
 def test_single_identity_invariance():
     n = 10
-    p = Rotation.random(n)
+    p = Rotation.random(n, random_state=0)
 
     result = p * Rotation.identity()
     assert_array_almost_equal(p.as_quat(), result.as_quat())
@@ -1008,7 +1008,7 @@ def test_align_vectors_no_rotation():
 def test_align_vectors_no_noise():
     rnd = np.random.RandomState(0)
     c = Rotation.from_quat(rnd.normal(size=4))
-    b = np.random.normal(size=(5, 3))
+    b = rnd.normal(size=(5, 3))
     a = c.apply(b)
 
     est, rmsd = Rotation.align_vectors(a, b)
@@ -1046,14 +1046,14 @@ def test_align_vectors_noise():
     rnd = np.random.RandomState(0)
     n_vectors = 100
     rot = Rotation.from_euler('xyz', rnd.normal(size=3))
-    vectors = np.random.normal(size=(n_vectors, 3))
+    vectors = rnd.normal(size=(n_vectors, 3))
     result = rot.apply(vectors)
 
     # The paper adds noise as independently distributed angular errors
     sigma = np.deg2rad(1)
     tolerance = 1.5 * sigma
     noise = Rotation.from_rotvec(
-        np.random.normal(
+        rnd.normal(
             size=(n_vectors, 3),
             scale=sigma
         )
@@ -1108,11 +1108,12 @@ def test_align_vectors_invalid_input():
 
 
 def test_random_rotation_shape():
-    assert_equal(Rotation.random().as_quat().shape, (4,))
-    assert_equal(Rotation.random(None).as_quat().shape, (4,))
+    rnd = np.random.RandomState(0)
+    assert_equal(Rotation.random(random_state=rnd).as_quat().shape, (4,))
+    assert_equal(Rotation.random(None, random_state=rnd).as_quat().shape, (4,))
 
-    assert_equal(Rotation.random(1).as_quat().shape, (1, 4))
-    assert_equal(Rotation.random(5).as_quat().shape, (5, 4))
+    assert_equal(Rotation.random(1, random_state=rnd).as_quat().shape, (1, 4))
+    assert_equal(Rotation.random(5, random_state=rnd).as_quat().shape, (5, 4))
 
 
 def test_slerp():
@@ -1246,8 +1247,8 @@ def test_multiplication_stability():
 
 
 def test_rotation_within_numpy_array():
-    single = Rotation.random()
-    multiple = Rotation.random(2)
+    single = Rotation.random(random_state=0)
+    multiple = Rotation.random(2, random_state=1)
 
     array = np.array(single)
     assert_equal(array.shape, ())

